src/main/java/seedu/duke/data/timetable/Timetable.java

```diff
@@ -39,17 +58,100 @@ public void setEvents(ArrayList<Event> events) {
+                yearlyEvents.add((YearlyEvent) event);
+            }
+        }
     }
 
     /**
      * Removes an event from the timetable.
      *
-     * @param event to be removed.
+     * @param index Index to be removed.
      */
-    public void deleteEvent(Event event) {
+    public Event deleteEvent(int index) throws SystemException {
+        if (index >= events.size() || index < 0) {
+            throw new SystemException(SystemException.ExceptionType.EXCEPTION_INDEX_OUT_OF_RANGE);
+        }
+        Event event = events.get(index);
+        events.remove(index);
+        if (event instanceof DailyEvent) {
+            dailyEvents.remove((DailyEvent) event);
+        } else if (event instanceof WeeklyEvent) {
+            weeklyEvents.remove((WeeklyEvent) event);
+        } else if (event instanceof MonthlyEvent) {
+            monthlyEvents.remove((MonthlyEvent) event);
+        } else if (event instanceof YearlyEvent) {
+            yearlyEvents.remove((YearlyEvent) event);
+        } else {
+            nonRecurringEvents.remove(event);
+        }
+        return event;
+    }
 
+    public int getDay() {
+        LocalDate today = LocalDate.now();
+        return today.getDayOfWeek().getValue();
     }
 
+    public int getDay(LocalDate date) {
+        return date.getDayOfWeek().getValue();
+    }
+
+    /**
+     * Gets the timetable for a specified month and year. Includes multiple recurrent events.
+     * @param year Year to check for timetable
+     * @param month Month to check for timetable
+     * @return A HashMap that maps the date to an ArrayList of all events that occurs on the day.
+     */
+    public HashMap<Integer, ArrayList<Event>> getMonthTimetable(int year, int month) {
+        HashMap<Integer, ArrayList<Event>> monthTimetable = new HashMap<>();
+        LocalDate startDate = LocalDate.of(year, month, 1);
+        LocalDate endDate = startDate.withDayOfMonth(startDate.lengthOfMonth());
+        ArrayList<Event> eventSet = new ArrayList<>();
+        eventSet.addAll(getAllRecurringEvents(startDate, endDate, dailyEvents));
+        eventSet.addAll(getAllRecurringEvents(startDate, endDate, weeklyEvents));
+        eventSet.addAll(getAllRecurringEvents(startDate, endDate, monthlyEvents));
+        eventSet.addAll(getAllRecurringEvents(startDate, endDate, yearlyEvents));
+        eventSet.addAll(nonRecurringEvents);
 
+        int numDays = startDate.lengthOfMonth();
+        for (int i = 1; i <= numDays; i++) {
+            monthTimetable.put(i, new ArrayList<Event>());
+        }
+
+        for (Event event : eventSet) {
+            int date = event.getDate().getDayOfMonth();
+            ArrayList<Event> eventList = monthTimetable.get(date);
+            eventList.add(event);
+            monthTimetable.put(date, eventList);
+        }
+        return monthTimetable;
+    }
+
+    /**
+     * Given an ArrayList of events, get all events that will be occur will occur during the specified time period.
```

Comment:
Same as above + phrasing error.
```suggestion
     * Given an ArrayList of events, get all events that will occur during the specified time period.
     
```